src/Strategy/AbstractCollectionStrategy.php

```diff
@@ -43,6 +44,10 @@ public function setCollectionName(string $collectionName): void
 
     public function getCollectionName(): string
     {
+        if ($this->collectionName === null) {
+            throw new RuntimeException('Collection name has not been set.');
```

Comment:
From https://matthiasnoback.nl/2018/09/assertions-and-assertion-libraries/

> User input will indeed be a reason for functions to fail. But so are external failures in outgoing calls. If a function reaches out to the database to fetch an entity by its ID, then the entity may not exist (anymore) and the call will fail. Before you make that call to the database, you don't know yet if the function will fail or not. This is why language designers usually make a difference between LogicExceptions and RuntimeExceptions. They all extend from the generic Exception class, but their intent is different. A RuntimeException is a failure caused by external, unpredictable things: the database, the filesystem, the network, etc. A LogicException is a programming mistake. It shouldn't have happened, but somehow, somewhere, a programmer didn't use a function well.

```suggestion
            throw new LogicException('Collection name has not been set.');
```

This also applies below.